app/src/main/java/com/team3663/scouting_app/utility/Logger.java

```diff
@@ -200,6 +167,7 @@ private void WriteOutDataFile(File in_File_Data) {
 
         try {
             // Write out the data
+            assert fos_data != null;
```

Comment:
What does `assert` do?

Reply:
it short for "assertion".  I'm "asserting" that this is true (fos_data != null).  If it's not true, the app will crash with an error that the assertion wasn't true.  It's a common way to "prove" something is true that should always be true.  I added it to avoid a compiler warning and to avoid having to write an if block around it.  the way the code is written, it really won't be null here, so using assert should be fine.

Reply:
I knew what it stood for I just was confused about what it could do if it wasn't true. The app crashing doesn't sound great.

Reply:
the app would have crashed if we used it as a null anyways.
also, it's what Android Studio suggested to do to avoid the warning.

Reply:
like i said, it's not null at this point.  so it's more about not having the warning.  i don't know what java does, but it might be an assert/crash only when running it in debug mode.  (there are things like debug asserts, but i don't know about this one.